packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -1414,6 +1416,42 @@ flutter:
       });
 
     });
+
+    testWithoutContext('Symlink failures give developer mode instructions on recent versions of Windows', () async {
+      final MockPlatform platform = MockPlatform();
+      when(platform.isWindows).thenReturn(true);
+      final MockOperatingSystemUtils os = MockOperatingSystemUtils();
+      when(os.name).thenReturn('Microsoft Windows [Version 10.0.14972.1]');
+
+      const FileSystemException e = FileSystemException('', '', OSError('', 1314));
+
+      expect(() => handleSymlinkException(e, platform: platform, os: os),
+        throwsToolExit(message: 'start ms-settings:developers'));
+    });
+
+    testWithoutContext('Symlink failures instruct developers to run as administrator on older versions of Windows', () async {
+      final MockPlatform platform = MockPlatform();
+      when(platform.isWindows).thenReturn(true);
+      final MockOperatingSystemUtils os = MockOperatingSystemUtils();
+      when(os.name).thenReturn('Microsoft Windows [Version 10.0.14393]');
+
+      const FileSystemException e = FileSystemException('', '', OSError('', 1314));
+
+      expect(() => handleSymlinkException(e, platform: platform, os: os),
+        throwsToolExit(message: 'administrator'));
+    });
+
+    testWithoutContext('Symlink failures only give instructions for specific errors', () async {
+      final MockPlatform platform = MockPlatform();
+      when(platform.isWindows).thenReturn(true);
+      final MockOperatingSystemUtils os = MockOperatingSystemUtils();
+      when(os.name).thenReturn('Microsoft Windows [Version 10.0.14393]');
+
+      const FileSystemException e = FileSystemException('', '', OSError('', 999));
+
+      // This should now throw.
+      handleSymlinkException(e, platform: platform, os: os);
```

Comment:
// This should not throw.

Alternatively if you want to be more explicit you could use:

`expect(() => handleSymlinkException(e, platform: platform, os: os), returnsNormally)`

Reply:
Ah, perfect. I tried to find an example of a similar test to see what the best construction was, but didn't find the right search terms.